Content.Server/Traits/Assorted/EmotionLoopComponent.cs

```diff
@@ -0,0 +1,21 @@
+using System.Numerics;
+
+namespace Content.Server.Traits.Assorted;
+
+/// <summary>
+/// This component allows triggering any emotion at random intervals.
+/// </summary>
+[RegisterComponent, Access(typeof(EmotionLoopSystem))]
+public sealed partial class EmotionLoopComponent : Component
+{
+    /// <summary>
+    /// A random interval between emotions (minimum, maximum).
+    /// </summary>
+    [DataField("timeBetweenEmotions", required: true)]
+    public Vector2 TimeBetweenEmotions { get; private set; }
+
+    [DataField(required: true)]
+    public string Emotion { get; private set; }
+
+    public float NextIncidentTime;
```

Comment:
Would it be better to use a `TimeSpan`? You can randomly generate a TimeSpan between a range, and you can just subtract that TimeSpan from the current time to see if you've passed it or not.